cli/src/commands/git/push.rs

```diff
@@ -295,17 +300,56 @@ pub fn cmd_git_push(
         return Ok(());
     }
 
-    validate_commits_ready_to_push(ui, &bookmark_updates, &remote, &tx, args)?;
+    let sign_behavior = if tx.settings().get_bool("git.sign-on-push")? {
+        Some(SignBehavior::Own)
+    } else {
+        None
+    };
+    let commits_to_sign =
+        validate_commits_ready_to_push(ui, &bookmark_updates, &remote, &tx, args, &sign_behavior)?;
+    let num_updated_signatures = commits_to_sign.len();
+
+    if args.dry_run {
+        if let Some(mut formatter) = ui.status_formatter() {
+            if num_updated_signatures > 0 {
+                writeln!(
+                    formatter,
+                    "Dry-run requested, skipped updating signatures of {num_updated_signatures} \
+                     commits"
+                )?;
+            }
```

Comment:
I think we can skip this message. What do you think? Then we should not have to duplicate the next two lines either, as long as we skip the call to `sign_commits_before_push()` in dry-run mode. I think the `sign_behavior` variable can be replaced by a `should_sign: bool` now that we only use `SignBehavior::Own`. Just an idea, possibly a bad one, so you decide.